From 970b984b58545655f3e81fc9c0d37578792493cc Mon Sep 17 00:00:00 2001
From: nibon7 <nibon7@gmail.com>
Date: Thu, 22 May 2025 17:11:23 +0800
Subject: [PATCH] shell: Make sure that all child widgets of PpsView are
 removed when closing document

When opening a document that contains multiple text form fields,
activate one form field, then another, and then exit without saving the
document, the last one doesn't unparent from PpsView.

```console
(papers:790001): Gtk-WARNING **: 17:14:19.400: Finalizing PpsView 0x55add05fdaa0, but it still has children left:
   - GtkTextView 0x55add12e81a0
```
---
 libview/pps-view.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libview/pps-view.c b/libview/pps-view.c
index b12b7bb1d682..8b680b44bed4 100644
--- a/libview/pps-view.c
+++ b/libview/pps-view.c
@@ -6233,6 +6233,8 @@ pps_view_dispose (GObject *object)
 	PpsView *view = PPS_VIEW (object);
 	PpsViewPrivate *priv = GET_PRIVATE (view);
 
+	pps_view_remove_all (view);
+
 	if (priv->model) {
 		g_signal_handlers_disconnect_by_data (priv->model, view);
 		g_clear_object (&priv->model);
-- 
2.49.0

